zendesk_attachment_backup.py

```diff
@@ -69,37 +73,51 @@ def collect_urls(input_files, print_urls, **kwargs):
 
    # Print URLs to file if requested
    if print_urls:
-      with open('attachment_urls.txt', "w") as fh:
+      target_file=os.path.join(kwargs.get("output_dir"), 'attachment_urls.txt')
+      with open(target_file, "w") as fh:
          for url in urls:
             fh.write("{}\n".format(url))
-      print("Printed results to attachment_urls.txt")
+      if not kwargs..get('quiet'):
+          print("Printed results to {}.txt".format(target_file))
 
    return downloads
 
-def download_files(downloads, output_dir='downloads', force_overwrite=False, **kwargs):
-   """ Save files to disk. Input: Dict with desired filename as key, URL as value.
-   """
-   # Make the directory if we need to
-   if not os.path.exists(output_dir):
-      os.makedirs(output_dir)
+def download_files(downloads, output_dir, force_overwrite, **kwargs):
+    """ Save files to disk. Input: Dict with desired filename as key, URL as value.
+    """
+    # Make the directory if we need to
+    if not os.path.exists(output_dir):
+        os.makedirs(output_dir)
    
-   # Check if any files exist and skip them if so
-   num_dls = len(downloads)
-   if not force_overwrite:
-      downloads = {fn:url for fn,url in downloads.iteritems() if not os.path.exists(os.path.join(output_dir, fn))}
-      if num_dls != len(downloads):
-         print("Skipping {} files as already downloaded.".format(num_dls - len(downloads)))
+    # Check if any files exist and skip them if so
+    num_dls = len(downloads)
+    if not force_overwrite:
+        downloads = {fn:url for fn,url in downloads.iteritems() if not os.path.exists(os.path.join(output_dir, fn))}
+        if num_dls != len(downloads) and not kwargs.get('quiet'):
+            print("Skipping {} files as already downloaded.".format(num_dls - len(downloads)))
 
    # Loop through the downloads and get them one at a time
-   num_dls = len(downloads)
-   i = 1
-   for fn, url in downloads.iteritems():
-      path = os.path.join(output_dir, fn)
-      print("Downloading {} of {} - {}".format(i, num_dls, fn))
-      dl = urllib2.urlopen(url)
-      with open(path, 'wb') as fh:
-         fh.write(dl.read())
-      i += 1
+    num_dls = len(downloads)
+    i = 1
+    for fn, url in downloads.iteritems():
+        path = os.path.join(output_dir, fn)
+        if not kwargs.get('quiet'):
+            print("Downloading {} of {} - {}".format(i, num_dls, fn))
+    try:
+        dl = urllib2.urlopen(url)
+        with open(path, 'wb') as fh:
+            fh.write(dl.read())
+    except urllib2.URLError:
+        try:
+            dl = urllib2.urlopen(url)
+            with open(path, 'wb') as fh:
+                fh.write(dl.read())
+        except urllib2.URLError:
+            print("Error downloading {}".format(url))
+            sys.exit(1)
```

Comment:
That looks way better than mine, but I'll keep the urllib2.ulropen separated because I want to be 100% certain that it closes properly.